handlers.go

```diff
@@ -16,8 +16,25 @@ func GetBeers(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 }
 
 // GetBeer returns a beer from the cellar
-func GetBeer(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	// TODO
+func GetBeer(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	w.Header().Set("Content-Type", "application/json")
+
+	ID, err := strconv.Atoi(ps.ByName("id"))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(fmt.Sprintf("%s is not a valid Beer ID, it must be a number.", ps.ByName("id")))
+		return
+	}
+
+	for _, v := range Cellar {
+		if v.ID == ID {
+			json.NewEncoder(w).Encode(v)
+			return
+		}
+	}
+
+	w.WriteHeader(http.StatusNotFound)
+	json.NewEncoder(w).Encode(fmt.Sprintf("The beer you requested does not exist."))
```

Comment:
Awesome - just a tiny thing here, you don't need the call to fmt.Sprintf here since we're not plugging in any variables into the string (that's what the %s or %d is for). Just `Encode("The beer you requested does not exist.")` will do :)

Reply:
I won't excuse any of my earlier fuck-ups, but in this case it's 4:30AM ^_^